visualization/app/codeCharta/ui/attributeSideBar/attributeSideBar.component.spec.ts

```diff
@@ -39,11 +39,8 @@ describe("AttributeSideBarController", () => {
 	}
 
 	function withMockedCodeMapPreRenderService() {
-		codeMapPreRenderService = attributeSideBarController["codeMapPreRenderService"] = jest.fn<CodeMapPreRenderService>(() => {
-			return {
-				getRenderFileMeta: jest.fn().mockReturnValue({ fileName: "my_fileName" })
-			}
-		})()
+		codeMapPreRenderService.getRenderFileMeta = jest.fn().mockReturnValue({ fileName: "my_fileName" })
+		attributeSideBarController["codeMapPreRenderService"] = codeMapPreRenderService
```

Comment:
I suggest to add a [linter rule](https://eslint.org/docs/rules/dot-notation) to use the dot notation in the future (if you agree, in a separate PR).

Reply:
Accessing private fields or functions is quite useful when writing tests or setting up a test.
We could enable that rule for our production code for sure, but tests shouldn't be included.

Reply:
Using the dot notation is independent from accessing private fields :)

Reply:
How? When I want to access a private field, I have to use `class["private_field"]`?  instead of `class.private_field`

Reply:
TIL. I was not aware that TypeScript used this specific notation to access private fields. IMO it's a bad practice to access any private field. I do not want to go into that discussion here though. I would actively mark access like these (by adding a comment) that it's a private properly that is intentionally accessed and that it's the TypeScript way to do so.

Calling something a private field that is still accessible is somewhat weird. The reason that it's working like that was AFAIK that it was difficult to create actual private members when TypeScript came up. That changed in the meanwhile and now JS has actual private fields that do not allow any access from outside.

I personally always test internals by either: using the dependency injection pattern or by just plainly checking the overall output of a higher level API that should only work if the internal path is hit. If the coverage does not show the expected lines to be covered, something's fishy.

----------------------

To get back to the actual comment: for now it would be possible to exclude tests for such a rule. On the long term, I'd try to refactor the code to the dependency injection pattern.

Reply:
We could use the [typescript-es-lint rule](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/dot-notation.md). It has an option to allow private class member access like this.

I'm not sure if you can allow/disallow individual rules. I think you can just define a pattern that applies to all rules.

Reply:
It is possible to enable specific rules for specific files by using [overrides](https://eslint.org/docs/user-guide/configuring#specifying-processor). It should be sufficient to just use the ["native" rule](https://eslint.org/docs/rules/dot-notation).